hw/cdc/tools/meridiancdc/run-cdc.tcl

```diff
@@ -0,0 +1,76 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Script to run MeridianCDC analysis (specific to OpenTitan but independent of the top-level design)
+
+proc get_env_var {name} {
+  if {[info exists ::env($name)]} {
+    set val "[set ::env([set name])]"
+    puts "::env($name) = $val"
+    return $val
+  } else {
+    puts "ERROR: Script run without $name environment variable."
+    quit
+  }
+}
+
+set FOUNDRY_ROOT        ""
+set SV_FLIST            [get_env_var "SV_FLIST"]
+set BUILD_DIR           [get_env_var "BUILD_DIR"]
+set DUT                 [get_env_var "DUT"]
+set CONSTRAINT          [get_env_var "CONSTRAINT"]
+set PARAMS              [get_env_var "PARAMS"]
+set ENV_FILE            [get_env_var "ENV_FILE"]
+
+# Used to disable some SDC constructs that are not needed by CDC.
+set IS_CDC_RUN 1
+
+# TODO(#24208): Increasing the `ri_max_total_range_bits` setting as below can help removing
+# black-box expressions, but it requires > 256 GiB of RAM, which is probably not available on
+# typical workstations.  We might want to enable this based on an environment variable.
+# set ri_max_total_range_bits 20
+
+set DEFINE ""
+if {$DEFINE != ""} {
+  analyze -sverilog +define+${DEFINE} -f ${SV_FLIST}
+} else {
+  analyze -sverilog -f ${SV_FLIST}
+}
+
+# Black-boxing prim_clock_div, prim_clock_gating, and prim_flash because each of them has a
+# tech-specific implementation, so analyzing RTL that won't be used on the chip is meaningless and
+# without this black box, MeridianCDC reports conflicts between how the design drives derived clocks
+# and how derived clocks are specified in the SDC.
+# TODO(#24208): ast_clks_byp should not be black-boxed, however.
+set BLACK_BOX_MODULES [list \
+  ast_clks_byp \
+  prim_clock_div \
+  prim_clock_gating \
+  prim_flash \
+]
+
+if {$PARAMS != ""} {
+  elaborate -params "$PARAMS" -black_box $BLACK_BOX_MODULES $DUT
+} else {
+  elaborate -black_box $BLACK_BOX_MODULES $DUT
+}
+
+set_user_cntl_synchronizer -name ot_cntl_synchronizer prim_flop_2sync
+set_user_reset_synchronizer -name ot_reset_synchronizer prim_rst_sync
+
+create_scenario { sdc env }
+
+set envs env_from_public_sdc.env
+
+read_sdc -scenario sdc -output_env $envs $CONSTRAINT
+
+current_scenario env
+read_env $envs
+
+touch $ENV_FILE
+read_env $ENV_FILE
+
+analyze_intent
+
+verify_cdc
```

Comment:
I am obviously no expert but comparing a bit the flow files for Verix and Meridian, it seems that there are some steps missing in the new flow like:
- setting the top modules
- reading in waivers
- writing out the report (I am aware that the parsing is currently missing and I believe that's fine at this point)

Does it make sense to add placeholders in the tcl script now (as the Verix reference is removed with this PR) or is adding these things anyway very obvious and tracked in an issue. What is your view?

Reply:
Thanks for these inputs!

- Setting the top modules: As we run CDC on the full chip (`chip_earlgrey_asic`), we don't need to set top modules other than `DUT` (which we do). The previous flow would have allowed to run CDC at the block level, which we currently don't intend to do, because most CDCs are between HW IP blocks.
- Reading in waivers: Currently there are no waivers, but at some point (and for generality), I agree that it would be useful to support loading a waiver file in the run script. I'll add it.
- Writing out reports: Right, I could as well add this to this PR already.